app/controllers/profiles_controller.rb

```diff
@@ -51,6 +51,11 @@ def search_profiles(params)
       all = all.merge(in_range_profiles)
     end
     
+    unless params["proximity"].blank? && params["user_location"].blank?
+      in_range_profiles = Profile.near(params["user_location"], params["proximity"], units: :km)
+      all = all.merge(in_range_profiles)
```

Comment:
You don't need to merge here. `near` is a chainable method like `where`, so you'd just be doing 
`all = all.near(params[:user_location], etc, etc)`